cmd/migrate/list_test.go

```diff
@@ -0,0 +1,1306 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+)
+
+// assertConfigsEquivalent checks if the transformed config contains all expected resources
+func assertConfigsEquivalent(t *testing.T, actual string, expected []string) {
+	// Parse the actual output to get structured data
+	actualFile, diags := hclwrite.ParseConfig([]byte(actual), "actual.tf", hcl.InitialPos)
+	if diags.HasErrors() {
+		t.Fatalf("Failed to parse actual output: %v", diags)
+	}
+
+	// Build a map of actual resources for easier comparison
+	actualResources := extractResources(actualFile.Body())
+
+	// For each expected resource, verify it exists with the right attributes
+	for i, exp := range expected {
+		expFile, diags := hclwrite.ParseConfig([]byte(exp), "expected.tf", hcl.InitialPos)
+		if diags.HasErrors() {
+			t.Fatalf("Failed to parse expected output %d: %v", i, diags)
+		}
+
+		// Extract expected resources
+		expResources := extractResources(expFile.Body())
+
+		// Check each expected resource
+		for expKey, expRes := range expResources {
+			actRes, found := actualResources[expKey]
+			if !found {
+				t.Errorf("Expected resource %s not found in output.\n\n=== Expected ===\n%s\n\n=== Actual Resources ===\n%s",
+					expKey, exp, formatResources(actualResources))
+				continue
+			}
+
+			// Compare attributes (order doesn't matter)
+			if !compareResourceAttributes(expRes, actRes) {
+				t.Errorf("Resource %s has different attributes.\n\n=== Expected ===\n%s\n\n=== Actual ===\n%s",
+					expKey, formatResource(expKey, expRes), formatResource(expKey, actRes))
+			}
+		}
+	}
+}
+
+// ResourceInfo holds information about a resource
+type ResourceInfo struct {
+	Type       string
+	Name       string
+	Attributes map[string]string
+}
+
+// extractResources extracts all resources from an HCL body
+func extractResources(body *hclwrite.Body) map[string]*ResourceInfo {
+	resources := make(map[string]*ResourceInfo)
+
+	for _, block := range body.Blocks() {
+		if block.Type() == "resource" && len(block.Labels()) >= 2 {
+			resType := block.Labels()[0]
+			resName := block.Labels()[1]
+			key := resType + "." + resName
+
+			attrs := make(map[string]string)
+			for name, attr := range block.Body().Attributes() {
+				// Convert attribute to string representation
+				tokens := attr.Expr().BuildTokens(nil)
+				attrs[name] = tokensToString(tokens)
+			}
+
+			resources[key] = &ResourceInfo{
+				Type:       resType,
+				Name:       resName,
+				Attributes: attrs,
+			}
+		}
+	}
+
+	return resources
+}
+
+// tokensToString converts HCL tokens to a normalized string
+func tokensToString(tokens hclwrite.Tokens) string {
```

Comment:
some useful thing that I found:

---

you can actually inline this function by just calling on `thing`

`string(thing.BuildTokens(nil).Bytes())`